docs: added scaffolding, config, proto and frontend (#891)
tendermint/starport: docs/7 Frontend/1 Overview.md

# Frontend Overview

> When scaffolding a blockchain, Starport also creates a Vue frontend application in `vue` directory. If this directory exists, `starport serve` automatically runs `npm run serve`.

Reviewer: ```suggestion
A Vue frontend app is created in `vue` directory when a blockchain is scaffolded. If the `vue` directory exists, the `starport serve` command automatically runs `npm run serve`.
```

Reviewer: @fadeev what  `npm run serve` do? why only for existing directories?

Author: `npm run serve` runs the frontend. We execute this command only if `vue` exists, otherwise there is nothing to run on the frontend.

> The frontend app uses packages from [`tendermint/vue`](https://github.com/tendermint/vue).

Reviewer 2: Should we explicitly name the packages? i.e. something like:

The front-end app is built using the @starport/vue and @starport/vuex packages. More information and documentation can be found at the [`tendermint/vue`](https://github.com/tendermint/vue) monorepo.

Author: Sounds good, thanks!

> The frontend app uses packages from [`tendermint/vue`](https://github.com/tendermint/vue).

Author: ```suggestion
The frontend app is built using the `@starport/vue` and `@starport/vuex` packages. More information and documentation can be found in the [`tendermint/vue`](https://github.com/tendermint/vue) monorepo.
```

> The frontend app uses packages from [`tendermint/vue`](https://github.com/tendermint/vue).

Reviewer: ```suggestion
The frontend app is built using  the @starport/vue and @starport/vue packages from the [`tendermint/vue`](https://github.com/tendermint/vue) monorepo.
```

> The frontend app is built using the `@starport/vue` and `@starport/vuex` packages. More information and documentation can be found in the [`tendermint/vue`](https://github.com/tendermint/vue) monorepo.

Reviewer: ```suggestion
The frontend app is built using the `@starport/vue` and `@starport/vuex` packages. For more details, see the [Monorepo for Starport front-end development](https://github.com/tendermint/vue) monorepo.
```

> ## Client code generation

Reviewer: ```suggestion
## Client Code Generation
```

> Starport automatically generates JavaScript, TypeScript and Vuex clients for your blockchain both for custom and standard Cosmos SDK modules. To enable client code generation add the following to `config.yml`:

Reviewer: ```suggestion
JavaScript (JS), TypeScript (TS), and Vuex clients are automatically generated for your blockchain for custom and standard Cosmos SDK modules. To enable client code generation, add the `client` entries to `config.yml`:
```

```yaml
client:
  vuex:
    path: "js"
```

> Starport will generate a Vuex client inside `js` directory. JS and TS clients are also generated, because they are dependencies of the Vuex client.

Reviewer: ```suggestion
A Vuex client is generated in the `js` directory. JS and TS clients are also generated because they are dependencies of the Vuex client.
```

> By default Starport watches the filesystem and regenerates clients for your custom modules automatically. Clients for standard Cosmos SDK modules are generated once, when you scaffold a blockchain. To regenerate all clients (for both custom and stadard Cosmos SDK modules), run `starport serve --reset-once --rebuild-proto-once`.

Reviewer: ```suggestion
By default, the filesystem is watched and the clients are regenerated automatically. Clients for standard Cosmos SDK modules are generated once when you scaffold a blockchain. 

To regenerate all clients for custom and standard Cosmos SDK modules, run the `starport serve --reset-once --rebuild-proto-once` command.
```
